crates/vim/src/normal.rs

```diff
@@ -380,6 +392,32 @@ fn yank_line(_: &mut Workspace, _: &YankLine, cx: &mut ViewContext<Workspace>) {
     })
 }
 
+fn save_selection_starts(editor: &Editor, cx: &mut ViewContext<Editor>) -> HashMap<usize, Anchor> {
+    let (map, selections) = editor.selections.all_display(cx);
+    selections
+        .iter()
+        .map(|selection| {
+            (
+                selection.id,
+                map.display_point_to_anchor(selection.start, Bias::Right),
+            )
+        })
+        .collect::<HashMap<_, _>>()
+}
+
+fn restore_selection_cursors(
+    editor: &mut Editor,
+    cx: &mut ViewContext<Editor>,
+    positions: &mut HashMap<usize, Anchor>,
+) {
+    editor.change_selections(Some(Autoscroll::fit()), cx, |s| {
+        s.move_with(|map, selection| {
+            let anchor = positions.remove(&selection.id).unwrap();
+            selection.collapse_to(anchor.to_display_point(map), SelectionGoal::None);
```

Comment:
Since we now have the code in two separate functions and someone _could_ (yes, unlikely) pass in another HashMap here, I think we should do a `if let Some(...) = ...` instead of `.unwrap()`. I get nervous whenever I see `.unwrap()` :)

Reply:
Note regarding our discussion about the cursor behavior : the solution using anchors, in its simplest form (I.e. saving & restoring selection start) differs from both vim and nvim. Now the cursors stays on the same column in the upper line.

Reply:
Even on the first indent and not just on repeat with `.`?

Reply:
Yes

Reply:
The behavior it gives us seems ok to me (and is better than it was before!). Does it feel right to you? If so, happy to merge.

Looking a bit more closely, I think:

* nvim resets the cursor to the column that was the "end" of the visual selection in the top row
* vim resets the cursor to the first non-whitespace in the top row
* zed leaves the cursor attached to the character at the top of the selection.

I think this is fine for now (and the zed behavior makes most sense to me, followed by vim, followed by nvim which makes the least sense). If there are future travelers who want a bit more vim compatibility here, I'd be open to that too, but given that vim and nvim disagree, I think we're ok to do our own thing.

Reply:
Ok for me.

Reply:
Also, I noticed an odd thing with visual line mode : selection is extending to the end of the previous line.